src/components/space/SyncScreen.js

```diff
@@ -0,0 +1,301 @@
+import Loader from '../common/Loader';
+import { clearSpace, getSpace, syncSpace } from '../../actions';
+import './SpaceScreen.css';
+import Styles from '../../Styles';
+import { HOME_PATH } from '../../config/paths';
+import SpaceNotFound from './SpaceNotFound';
+import { generateGetSpaceEndpoint } from '../../config/endpoints';
+import { DEFAULT_GET_REQUEST } from '../../config/rest';
+import {
+  ERROR_MESSAGE_HEADER,
+  ERROR_SYNCING_MESSAGE,
+} from '../../config/messages';
+
+const SELECTED_SPACE_PROPERTIES = ['name', 'image', 'description', 'phases'];
+
+const diffEditorStyles = {
+  variables: {
+    highlightBackground: '#fefed5',
+    highlightGutterBackground: '#ffcd3c',
+  },
+  line: {
+    padding: '10px 2px',
+    wordBreak: 'break-word',
+    '&:hover': {
+      background: '#a26ea1',
+    },
+  },
+  titleBlock: {
+    pre: {
+      fontSize: '16px',
+      fontWeight: 'bold',
+      fontFamily: 'arial',
+    },
+  },
+};
+
+class SyncScreen extends Component {
+  state = {
+    remoteSpace: null,
+    openDrawer: false,
+  };
+
+  static propTypes = {
+    space: ImmutablePropTypes.contains({
+      id: PropTypes.string,
+      description: PropTypes.string,
+    }).isRequired,
+    dispatchGetSpace: PropTypes.func.isRequired,
+    dispatchClearSpace: PropTypes.func.isRequired,
+    dispatchSync: PropTypes.func.isRequired,
+    activity: PropTypes.bool.isRequired,
+    deleted: PropTypes.bool.isRequired,
+    classes: PropTypes.shape({
+      root: PropTypes.string.isRequired,
+      appBar: PropTypes.string.isRequired,
+      appBarShift: PropTypes.string.isRequired,
+      drawer: PropTypes.string.isRequired,
+      drawerPaper: PropTypes.string.isRequired,
+      drawerHeader: PropTypes.string.isRequired,
+      content: PropTypes.string.isRequired,
+      menuButton: PropTypes.string.isRequired,
+      contentShift: PropTypes.string.isRequired,
+      hide: PropTypes.string.isRequired,
+      button: PropTypes.string.isRequired,
+    }).isRequired,
+    theme: PropTypes.shape({ direction: PropTypes.string.isRequired })
+      .isRequired,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        id: PropTypes.string.isRequired,
+      }),
+    }).isRequired,
+    location: PropTypes.shape({
+      search: PropTypes.string.isRequired,
+    }).isRequired,
+    history: PropTypes.shape({
+      goBack: PropTypes.func.isRequired,
+      push: PropTypes.func.isRequired,
+      replace: PropTypes.func.isRequired,
+    }).isRequired,
+    t: PropTypes.func.isRequired,
+  };
+
+  async componentDidMount() {
+    const {
+      match: {
+        params: { id },
+      },
+      history: { goBack },
+      location: { search },
+      dispatchGetSpace,
+    } = this.props;
+
+    // tell action creator if this space has already been saved
+    const { saved } = Qs.parse(search, { ignoreQueryPrefix: true });
+    dispatchGetSpace({ id, saved: saved === 'true' });
+
+    // get remote space
```

Comment:
Why is all of this here? Why don't you use an action creator. I think we already have one to fetch a remote space.